deploy-terraform.sh

```diff
@@ -0,0 +1,80 @@
+#!/bin/bash 
+# script to deploy terraform
+# exit on error or if a variable is unbound
+set -eu
+
+# create a tag from travis variables
+# this will be the docker tag that gets deployed
+TAG=$TRAVIS_BUILD-$TRAVIS_COMMIT
+
+# set env vars depending on which branch we are on
+# TF_VAR_variable_name gets passed to `terraform`
+# to set variables.
+# The variables starting PROD or STAGING should be exported
+# from your running environment
+
+if [ "$TRAVIS_BRANCH" = master ]
+then
+  echo "creating production env vars for terraform"
+  export TF_VAR_docker_image=$TAG
+  export TF_VAR_environment=prod
+  export TF_VAR_secret_key_base=$PROD_SECRET_KEY_BASE
```

Comment:
Are these variables are being loaded from the Travis context and we've added them there manually? 

If so, I'm wondering about what we want to happen if a key is mistakenly mispelt or a omitted. The impact is that our app starts breaking in odd ways and Rollbar reports a problem. Is there something we could do here that checks for the presence of a value for each of these variables and aborts with a meaningful error like "missing environment variable?" if one is at least missing?

Reply:
`set -eu` causes the script to error and exit if any are missing.